File: 9781430238010/9781430238010_Ch13/TstEql.c
#include <stdio.h>

int main(int argc, char **argv)
{
	char myArray[] = {'a','b','c','d'};
	int pos = -1;
	char c = 'c';
	int j, length = 4;

	for (j = 0; j < length; j++) {
		if (c = myArray[j]) {
			pos = j;
			break;
		}
	}
	printf("%d\n",pos);
}

File: 9781430238010/9781430238010_Ch13/TstEql.java
/**
 * This is the Java version of the TstEql program
 * that deliberately makes the == vs = mistake.
 * Unlike the C program, this program will not compile.
 * The objective is to see what the error message is
 */

public class TstEql
{
	public static void main(String [] args)
	{
		char [] myArray = {'a','b','c','d'};
		int pos = -1;
		char c = 'c';

		for (int j = 0; j < myArray.length; j++) {
			if (c = myArray[j]) {
				pos = j;
				break;
			}
		}
		System.out.println(pos);
	}
}

